tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java

```diff
@@ -67,12 +68,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             out.println("<b>Parts:</b>&nbsp;" + parts.size());
             for (Part p : parts)
             {
-                out.println("<h3>" + p.getName() + "</h3>");
+                out.println("<h3>" + StringUtil.sanitizeXmlString(p.getName()) + "</h3>");
                 out.println("<b>Size:</b>&nbsp;" + p.getSize());
                 if (p.getContentType() == null || p.getContentType().startsWith("text/plain"))
                 {
                     out.println("<p>");
-                    IO.copy(p.getInputStream(), out);
+                    String content = IO.toString(p.getInputStream());
```

Comment:
could this be a problem with a large content size

Reply:
Yes - we should probably limit it echoing the first 32KB (ie size of response buffer).